sound/soc/sof/control.c

```diff
@@ -458,15 +475,24 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 				    "error: bytes_ext put failed to resume %d\n",
 				    ret);
 		pm_runtime_put_noidle(sdev->dev);
+		kfree(new_cdata);
 		return ret;
 	}
 
 	/* notify DSP of byte control updates */
-	snd_sof_ipc_set_get_comp_data(sdev->ipc, scontrol,
-				      SOF_IPC_COMP_SET_DATA,
-				      SOF_CTRL_TYPE_DATA_SET,
-				      scontrol->cmd,
-				      true);
+	ret = snd_sof_ipc_set_get_comp_data(sdev->ipc, scontrol,
+					    SOF_IPC_COMP_SET_DATA,
+					    SOF_CTRL_TYPE_DATA_SET,
+					    scontrol->cmd,
+					    true);
+
+	/* If IPC was successful copy new data into PM cache */
```

Comment:
What is a 'PM cache'?
Also isn't the issue rather that the IPC return value was not handled properly? I don't see the added value of the memcpy?

Reply:
The cdata points to persistent memory for controls that I call "PM cache". With this patch the persistent memory is updated only if the IPC was successful.

Reply:
Sounds like using [RFC] was useful. I'll update the comments to something better. @ranj063 What do you think about this approach?

Reply:
@singalsu when will you get a -EBUSY from the DSP?

Reply:
and does it mean you cannot apply a new config with sof-eqctl while playback is in progress?

Reply:
> Sounds like using [RFC] was useful. I'll update the comments to something better. @ranj063 What do you think about this approach?

yes, please. using words like 'cache corruption' will raise tons of questions, when what you are really doing is making sure the data passed by userspace is only cached by the kernel for future suspend/resume transitions if the initial IPC succeeds.